include/d/d_bg_s_gnd_chk.h

```diff
@@ -67,14 +67,17 @@ class dBgS_ObjGndChk_All : public dBgS_ObjGndChk {
 
 class dBgS_CamGndChk : public dBgS_GndChk {
 public:
-    dBgS_CamGndChk() { /* SetCam(); */ }
+    dBgS_CamGndChk() { SetCam(); }
 
     virtual ~dBgS_CamGndChk() {}
 };
 
 class dBgS_CamGndChk_Wtr : public dBgS_CamGndChk {
 public:
-    dBgS_CamGndChk_Wtr();
+    dBgS_CamGndChk_Wtr() {
+        OffNormalGrp();
+        OnWaterGrp();
+    };
 
     virtual ~dBgS_CamGndChk_Wtr();
```

Comment:
```suggestion
    virtual ~dBgS_CamGndChk_Wtr() {}
```
Is this what you meant about dBgS causing functions to not match?

Reply:
In functions like [this](https://decomp.me/scratch/F7YHk) it seems like there's some missing vtable functions that get called as part of `dBgS_CamGndChk_Wtr`'s dtor. Also the extra `addi` instruction might be something wrong with the ctor? Not too sure if that's what's going on here

Reply:
https://decomp.me/scratch/OvRLs
The vtable function's aren't being called, it's just the destructor. You haven't implemented the destructor so it can't be inlined, which is why all the vtable stuff is missing from that scratch